databricks_cli/instance_pools/cli.py

```diff
@@ -0,0 +1,157 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from json import loads as json_loads
+
+import click
+from tabulate import tabulate
+
+from databricks_cli.click_types import OutputClickType, JsonClickType, InstancePoolIdClickType
+from databricks_cli.instance_pools.api import InstancePoolsApi
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, pretty_format, json_cli_base, \
+    truncate_string
+from databricks_cli.configure.config import provide_api_client, profile_option, debug_option
+from databricks_cli.version import print_version_callback, version
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None, type=click.Path(),
+              help='File containing JSON request to POST to /api/2.0/instance_pools/create.')
+@click.option('--json', default=None, type=JsonClickType(),
+              help=JsonClickType.help('/api/2.0/instance_pools/create'))
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_cli(api_client, json_file, json):
+    """
+    Creates a Databricks instance pool.
+
+    The specification for the request json can be found at
+    https://docs.databricks.com/api/latest/instance-pools.html#create
+    """
+    json_cli_base(json_file, json, lambda json: InstancePoolsApi(api_client).create_instance_pool(json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None, type=click.Path(),
+              help='File containing JSON request to POST to /api/2.0/instance_pools/edit.')
+@click.option('--json', default=None, type=JsonClickType(),
+              help=JsonClickType.help('/api/2.0/instance_pools/edit'))
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def edit_cli(api_client, json_file, json):
+    """
+    Edits a Databricks instance pool.
+
+    The specification for the request json can be found at
+    https://docs.databricks.com/api/latest/instance-pools.html#edit
+    """
+    if not bool(json_file) ^ bool(json):
+        raise RuntimeError('Either --json-file or --json should be provided')
+    if json_file:
+        with open(json_file, 'r') as f:
+            json = f.read()
+    deser_json = json_loads(json)
+    InstancePoolsApi(api_client).edit_instance_pool(deser_json)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--instance-pool-id', required=True, type=InstancePoolIdClickType(),
+              help=InstancePoolIdClickType.help)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_cli(api_client, instance_pool_id):
+    """
+    Removes a Databricks instance pool given its ID.
+
+    The instance pool is removed asynchronously. Once the deletion has completed,
+
+    Use ``databricks instance_pools get --instance-pool-id INSTANCE_POOL_ID`` to check
```

Comment:
What termination states are there? I didn't see any in the API. BTW, this should be `databricks instance-pools`

Reply:
OK. There is no `termination state`. One can find following message if the pool is deleted:
```
Error: {"error_code":"RESOURCE_DOES_NOT_EXIST","message":"Can't find an instance pool with id: 0731-231749-bent3-pool-3UyhGBfP."}
```